src/Configuration.php

```diff
@@ -1818,23 +1912,23 @@ private static function retrieveShebang(stdClass $raw): ?string
 
     private static function retrieveSigningAlgorithm(stdClass $raw): int
     {
-        // TODO: trigger warning: if no signing algorithm is given provided we are not in dev mode
-        // TODO: trigger a warning if the signing algorithm used is weak
-        // TODO: no longer accept strings & document BC break
         if (false === isset($raw->algorithm)) {
-            return Phar::SHA1;
+            return self::DEFAULT_SIGNING_ALGORITHM;
```

Comment:
what if this algorithms is not one available in `get_phar_signing_algorithms` ?

Reply:
`get_phar_signing_algorithms()` would need to be updated (which is unlikely to happen often so maintaining that should be ok)